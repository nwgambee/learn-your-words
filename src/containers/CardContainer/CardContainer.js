import React, { Component } from 'react';
import { connect } from 'react-redux';
import VocabCard from '../VocabCard/VocabCard'
import './CardContainer.scss'
import PropTypes from 'prop-types'

 export const CardContainer = (props) => {
    return (
        <section className='card-container'>
            <VocabCard wordDetails={props.searchedWordDetails}/>
        </section>
    )
}

export const mapStateToProps = (state) => ({
    searchedWordDetails: state.searchedWord,
    vocabList: state.vocabList
  })
  
export default connect(mapStateToProps)(CardContainer)

CardContainer.propTypes = {
    // searchedWordDetails: PropTypes.object,
    vocabList: PropTypes.array
}