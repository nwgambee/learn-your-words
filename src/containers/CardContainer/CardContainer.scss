.card-container {
  height: 100vh;
  background-color: #882FF6;
  display: flex;
  align-content: center;
  justify-content: center;
  flex-direction: row;
}