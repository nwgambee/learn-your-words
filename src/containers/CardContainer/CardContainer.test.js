import React from 'react';
import { CardContainer, mapStateToProps, mapDispatchToProps } from './CardContainer';
import { shallow } from 'enzyme';



describe('CardContainer', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<CardContainer searchedWordDetails={{word: 'word', desc: 'this is a desc'}}/>);
    expect(wrapper).toMatchSnapshot();
  })
  describe('mapStateToProps', () => {
    it('should return an object with the searchedWordDetails and vocabList', () => {
      const mockState = {
        searchedWord: {word: 'hat', definition: 'this is a definition', type: 'noun'},
        loadingStatus: true,
        vocabList: [{word: 'hat'}, {word: 'cat'}],
        someOtherProp: false
      }
      const expected = {
        searchedWordDetails: {word: 'hat', definition: 'this is a definition', type: 'noun'},
        vocabList: [{word: 'hat'}, {word: 'cat'}],
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})
