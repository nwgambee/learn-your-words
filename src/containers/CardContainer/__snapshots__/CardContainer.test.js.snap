// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
<section
  className="card-container"
>
  <Connect(VocabCard)
    wordDetails={
      Object {
        "desc": "this is a desc",
        "word": "word",
      }
    }
  />
</section>
`;
